agent/agent.go

```diff
@@ -384,17 +384,9 @@ func (a *Agent) Start() error {
 
 	// Warn if the node name is incompatible with DNS
 	if InvalidDnsRe.MatchString(a.config.NodeName) {
-		a.logger.Warn("Node name will not be discoverable "+
-			"via DNS due to invalid characters. Valid characters include "+
-			"all alpha-numerics and dashes.",
-			"node_name", a.config.NodeName,
-		)
+		return fmt.Errorf("Invalid node name %q - contains invalid characters", a.config.NodeName)
 	} else if len(a.config.NodeName) > MaxDNSLabelLength {
-		a.logger.Warn("Node name will not be discoverable "+
-			"via DNS due to it being too long. Valid lengths are between "+
-			"1 and 63 bytes.",
-			"node_name", a.config.NodeName,
-		)
+		a.logger.Warn("Invalid node name %q node_name - exceeds max length of 63 characters", a.config.NodeName)
```

Comment:
Looking at where `InvalidDnsRe` and `MaxDNSLabelLength` are used I notice the 3 places updated in this PR are the only references outside the `agent/dns` package. I also noticed the new error messages are all quite similar.

What do you think about something like this:

Instead of exporting two constant from `agent/dns`, we could export a single function

```go
func ValidateName(name string) error { ... }
```

callers could wrap the error and add the the part of the error that is different:

```go
if err := dns.ValidateName(a.config.NodeName); err != nil {
    return errwrap.Errorf("Invalid node name {{err}}", err)
}
```

Reply:
Hey! I really like this idea. Good catch!